openpype/hosts/fusion/plugins/create/create_saver.py

```diff
@@ -149,10 +149,11 @@ def _collect_unmanaged_saver(self, tool):
         comp = tool.Comp()
 
         # Allow regular non-managed savers to also be picked up
-        project = legacy_io.Session["AVALON_PROJECT"]
-        asset = legacy_io.Session["AVALON_ASSET"]
-        task = legacy_io.Session["AVALON_TASK"]
+        project = self.create_context.get_current_project_name()
+        asset = self.create_context.get_current_asset_name()
+        task = self.create_context.get_current_task_name()
 
+        # TODO don't re-query asset document all the time
```

Comment:
Any pointers on how that could potentially be done?

Reply:
This method can expect project name, asset document and task name, all of them can be prepared before the tools loop and just passed in.